server/app/src/androidTest/java/androidx/test/uiautomator/UiScrollableCustom.java

```diff
@@ -24,6 +29,29 @@ public boolean scrollIntoView(UiSelector selector, boolean forceScroll) throws U
                 found = true;
             } else {
                 boolean scrolled = this.scrollForward(100);
+                getDevice().waitForIdle();
+                if (!forceScroll && !scrolled) break;
+            }
+        }
+        return found;
+    }
+
+    public boolean scrollIntoView(UiSelector selector, String targetBySelectorStrategy, boolean forceScroll) throws UiObjectNotFoundException, NoSuchMethodException, IllegalAccessException, InvocationTargetException, InvocationTargetException {
+        Tracer.trace(new Object[]{selector});
+        UiSelector childSelector = this.getSelector().childSelector(selector);
+        Method strategyMethod = By.class.getMethod(targetBySelectorStrategy, String.class);
+        BySelector byselector = (BySelector) strategyMethod.invoke(By.class, (String.valueOf(childSelector)));
+        boolean found = false;
```

Comment:
`found` value is not used anymore. First of all this method required `return` statement. Secondly you could use `found` as part of the `for` condition and avoid using `break`. This will make method more readable.

Reply:
Changes are adopted as per review comment